sympy/polys/numberfields/primes.py

```diff
@@ -396,7 +396,7 @@ def _compute_test_factor(p, gens, ZK):
     # predicts that such an element must exist, so nullspace should
     # be non-trivial.
     x = B.nullspace()[0, :].transpose()
-    beta = ZK.parent(ZK.matrix * x, denom=ZK.denom)
+    beta = ZK.parent(ZK.matrix * x.convert_to(ZZ), denom=ZK.denom)
```

Comment:
@skieffer this change preserves existing behaviour. Let me know if something else should be done here.

Reply:
Thanks for the question, and I'm sorry I wasn't able to respond in time. But your fix looks good. Thanks for this.

Reply:
As the author of the only code in the whole of SymPy that depended on the previous behaviour do you have any opinions about `GF(3).unify(ZZ)` and the general change in this PR that disallows it?

Reply:
Good question. I'll comment on the main thread, #25653.